### ISSUE TITLE
Squiz.PHP.DisallowMultipleAssignments false positive when using namespaces with static assignments

### ISSUE THREAD

**Issue by dbarr** [squizlabs/PHP_CodeSniffer#1949]
In the following snippet:

```
<?php

class A {
    public static $a;
}

A::$a = 'b'; // works
\A::$a = 'c'; // fails
\A\B\C::$d = 'e'; // fails
```

The last two assignments triggers _"Assignments must be the first block of code on a line (Squiz.PHP.DisallowMultipleAssignments.Found)"_ due to T_NS_SEPARATOR not being considered.

**Comment by gsherwood** [comment]
This was a tough one as it was quite an old sniff and required a rewrite to get this working. But all tests are passing and I think it is working now. Thanks for the bug report. The fix will be in the 3.3.0 version, but if you are able to test on your code base before release, it would be really helpful given the large change to the sniff.

**Comment by dbarr** [comment]
Not related to namespaces, but I noticed the error suppression operator (yep, cringe) causes the same sniff:

`@$a = 1;`

**Comment by dbarr** [comment]
Also, foreach without curly braces or semi-column:

```
$a = [];
foreach ($a as $b)
    $c = 'd';
```

**Comment by gsherwood** [comment]
Thanks a lot for reporting those as well. I've fixed them up.
